.github/workflows/build-test.yml

```diff
@@ -0,0 +1,134 @@
+name: build-test
+on:
+  push:
+    branches:
+      - releng/fix-circle-migration
+      - '!**-ui*'
+      - '!docs-*'
+      - '!backport/docs-*'
+      - '!stable-website'
```

Comment:
This should be `on: pull_request`, and `on: push` for the branches we've filtered for in test-core.yaml